builtin/providers/cloudstack/provider.go

```diff
@@ -72,10 +88,42 @@ func Provider() terraform.ResourceProvider {
 }
 
 func providerConfigure(d *schema.ResourceData) (interface{}, error) {
+	apiURL := d.Get("api_url").(string)
+	apiKey := d.Get("api_key").(string)
+	secretKey := d.Get("secret_key").(string)
+
+	if configFile := d.Get("config").(string); configFile != "" {
+		config, err := ini.Load(configFile)
+		if err != nil {
+			return config, err
+		}
+
+		section, err := config.GetSection(d.Get("profile").(string))
+		if err != nil {
+			return config, err
+		}
+
+		if apiURL == "" {
+			apiURL = section.Key("url").String()
+		}
+
+		if apiKey == "" {
+			apiKey = section.Key("apikey").String()
+		}
+
+		if secretKey == "" {
+			secretKey = section.Key("secretkey").String()
+		}
```

Comment:
As  explained in the comment above, I don't think we should add this option here. Let's make this as clear as possible and either use one or the other...